tests/e2e/ansible/install_test_deps.yaml

```diff
@@ -39,7 +39,7 @@
       ignore_errors: yes
     - name: Install kustomize
       shell: |
-        curl -s --retry 3 --retry-delay 10 -u ${USER}:${GITHUB_TOKEN} "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"  | bash
+        curl -s --retry 6 --retry-delay 10 -u $USER:$GITHUB_TOKEN "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"  | bash
```

Comment:
Is there a reason for the `${USER}` -> `$USER`? They should be just a different alternatives for the same.

As for the retry, sure. Having more attempts might help but then the commit message is not accurate. I'd suggest doing only the necessary change (retry) and mention in the commit message (Why&How) that in the CI we're hitting networking stability issues so this commit is increasing the retry count to up to 6 attempts to mitigate that. What do you think?

Reply:
@ldoktor right now this PR is on debug this is not the final PR yet. I am doing some testing as this is the only way that I can test using the GHA environment, that is why I am trying to analyze and reproduce the error....I have been researching and seems like we are not the only ones that we are being hit by this kind of issue...

Reply:
after reading I think the best solution is just to clone the repo and install...seems like other people have faced this issue on their projects too...and seems like even passing the Token is not enough

Reply:
Is the download the issue or the actual deployment? Perhaps we could bundle `known to work` version of the script.

Also to test it you might want to split the `curl` and `bash` parts into 2 steps to see which one is causing the troubles.

Reply:
@ldoktor the problem starts here https://github.com/kubernetes-sigs/kustomize/blob/master/hack/install_kustomize.sh#L141 and this is the error that we got https://github.com/kubernetes-sigs/kustomize/blob/master/hack/install_kustomize.sh#L145

Reply:
so even in our workflow we are passing the token seems like the error is still there randomly...